tests/test_everflow_testbed.py

```diff
@@ -0,0 +1,664 @@
+    tor_dest_ports_ptf_id = []
+    tor_dest_lag_name = []
+    get_port_info(tor_ports, tor_dest_ports, tor_dest_ports_ptf_id, tor_dest_lag_name)
+
+    spine_dest_ports = []
+    spine_dest_ports_ptf_id = []
+    spine_dest_lag_name = []
+    get_port_info(spine_ports, spine_dest_ports, spine_dest_ports_ptf_id, spine_dest_lag_name)
+
+    setup_information = {
+        'router_mac': host_facts['ansible_Ethernet0']['macaddress'],
+        'tor_ports': tor_ports,
+        'spine_ports': spine_ports,
+        'port_channels': port_channels,
+        'test_mirror_v4' : test_mirror_v4,
+        'test_mirror_v6' : test_mirror_v6,
+        'ingress' : {'ingress': test_ingress_mirror_on_ingress_acl, 'egress' : test_egress_mirror_on_ingress_acl},
+        'egress' : {'ingress': test_ingress_mirror_on_egress_acl, 'egress' : test_egress_mirror_on_egress_acl},
+        'tor' : {'src_port' : spine_ports[0], 'src_port_ptf_id' : str(mg_facts['minigraph_port_indices'][spine_ports[0]]), 'dest_port' : tor_dest_ports, 'dest_port_ptf_id':tor_dest_ports_ptf_id,
+                 'dest_port_lag_name': tor_dest_lag_name},
+        'spine' : {'src_port' : tor_ports[0], 'src_port_ptf_id' : str(mg_facts['minigraph_port_indices'][tor_ports[0]]), 'dest_port' : spine_dest_ports, 'dest_port_ptf_id':spine_dest_ports_ptf_id,
+                   'dest_port_lag_name': spine_dest_lag_name}
+    }
+
+
+    # This is important to add since for Polcier test case regular packet
+    # and mirror packet can go to same interface which cause tail drop of 
+    # police packet and test case cir/cbs calculation gets impacted
+    # We are making Regular Traffic has dedicated route and don't use
+    # default route.
+
+    peer_ip, peer_mac = get_neighbor_info(duthost, spine_dest_ports[3])
+    
+    add_route(duthost, "30.0.0.1/24", peer_ip)
+
+    yield setup_information
+    
+    remove_route(duthost, "30.0.0.1/24", peer_ip)
+
+
+
+@pytest.fixture
+def partial_ptf_runner(request, duthost, ptfhost):
+    def _partial_ptf_runner(setup_info, session_info, acl_stage, mirror_type,  expect_receive = True, test_name = None, **kwargs):
+        params = {
+                  'hwsku' :  duthost.facts['hwsku'],
+                  'asic_type' :  duthost.facts['asic_type'],
+                  'router_mac': setup_info['router_mac'],
+                  'session_src_ip' : session_info['session_src_ip'],
+                  'session_dst_ip' : session_info['session_dst_ip'],
+                  'session_ttl' : session_info['session_ttl'],
+                  'session_dscp' : session_info['session_dscp'],
+                  'acl_stage' : acl_stage,
+                  'mirror_stage' : mirror_type,
+                  'expect_received' : expect_receive }
+        params.update(kwargs)
+        ptf_runner(host=ptfhost,
+                   testdir="acstests",
+                   platform_dir="ptftests",
+                   testname="everflow_tb_test.EverflowTest" if test_name is None else test_name,
```

Comment:
Really minor nitpick, but this is the preferred way of doing this in python:
```suggestion
                   testname="everflow_tb_test.EverflowTest" if not test_name else test_name,
```

Reply:
@daall Thanks Updated.